youtrack_sdk/entities.py

```diff
@@ -142,29 +146,45 @@ class PeriodProjectCustomField(ProjectCustomField):
 
 
 class IssueCustomField(BaseModel):
-    id: Optional[str]
-    name: Optional[str]
-    project_custom_field: Optional[ProjectCustomFieldType] = Field(alias="projectCustomField")
+    id: Optional[str] = None
+    name: Optional[str] = None
+    project_custom_field: Optional[ProjectCustomFieldType] = Field(alias="projectCustomField", default=None)
 
 
 class TextIssueCustomField(IssueCustomField):
     type: Literal["TextIssueCustomField"] = Field(alias="$type", default="TextIssueCustomField")
-    value: Optional[TextFieldValue]
+    value: Optional[TextFieldValue] = None
 
 
 class SimpleIssueCustomField(IssueCustomField):
     type: Literal["SimpleIssueCustomField"] = Field(alias="$type", default="SimpleIssueCustomField")
-    value: Optional[DateTime | StrictStr | StrictInt | StrictFloat]
+    value: Optional[datetime | StrictStr | StrictInt | StrictFloat] = None
+
+    @field_validator("value")
```

Comment:
Explanation from the meeting: in the past this was in the type, here it is necessary, because the validation result depends on another field. Decision: @IamAbbey will check if it's possible to do it in V2 in the type. If not, then this way will be accepted.

Reply:
I am at the moment still unable to find possible way to do it in v2 to access other field value using Annotated Types or in type.

Reply:
https://github.com/pydantic/pydantic/issues/6732

Reply:
Bottom line: maybe they will add it, but until then, we can use model-bound validator.